pkg/pbflow/proto.go

```diff
@@ -99,13 +101,32 @@ func FlowToPB(fr *model.Record, s *ovnobserv.SampleDecoder) *Record {
 		pbflowRecord.Network.DstAddr = &IP{IpFamily: &IP_Ipv4{Ipv4: model.IntEncodeV4(fr.Id.DstIp)}}
 	}
 	if s != nil {
-		seen := make(map[string]bool)
+		seen := make(map[ovnmodel.NetworkEvent]bool)
+		pbflowRecord.NetworkEventsMetadata = make([]*NetworkEvent, 0)
 		for _, metadata := range fr.Metrics.NetworkEvents {
 			if !model.AllZerosMetaData(metadata) {
 				if md, err := s.DecodeCookie8Bytes(metadata); err == nil {
 					protoLog.Debugf("Network Events Metadata %v decoded Cookie: %v", metadata, md)
 					if !seen[md] {
-						pbflowRecord.NetworkEventsMetadata = append(pbflowRecord.NetworkEventsMetadata, md)
+						e := ovnmodel.ACLEvent{NetworkEvent: md}
```

Comment:
Also you don't have the guarantee that md is of type NetworkEvent
Should be something like:
```go

					if acl, ok := md.(*ovnmodel.ACLEvent); ok {
						// (fill pbEvent)
					} else {
                                               // something generic here, such as:
                                               pbEvent := NetworkEvent{
								Events: map[string]string{
								    "Message": md.String()
                                                                },
                                                          },
					}
```

Reply:
I don't think we will hit a case that will generate string but can't get cast as ACLEvent but that change won't hurt